src/main/java/ui/PanelMenuCreator.java

```diff
@@ -0,0 +1,133 @@
+                public void changed(ObservableValue<? extends Number> arg0, Number arg1, Number arg2) {
+                    for (Node child : panelsScrollPane.getChildrenUnmodifiable()) {
+                        if (child instanceof ScrollBar) {
+                            ScrollBar scrollBar = (ScrollBar) child;
+                            if (scrollBar.getOrientation() == Orientation.HORIZONTAL
+                                    && scrollBar.visibleProperty().get()) {
+                                panelControl.scrollToCurrentlySelectedPanel();
+                                break;
+                            }
+                        }
+                    }
+                    panelControl.widthProperty().removeListener(this);
+                }
+            };
+            panelControl.widthProperty().addListener(listener);
+        });
+        createRight.setAccelerator(CREATE_RIGHT_PANEL);
+        return createRight;
+    }
+
+    public MenuItem closePanel(){
+        MenuItem closePanel = new MenuItem("Close");
+        closePanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Close");
+            panelControl.closeCurrentPanel();
+        });
+        closePanel.setAccelerator(CLOSE_PANEL);
+        return closePanel;
+    }
+
+    public MenuItem createAssigneePanel(){
+        MenuItem assigneePanel = new MenuItem("Self-assigned issues");
+        assigneePanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Create Self-assigned issues panel");
+            generatePanelWithNameAndFilter("Open issues and PR's",
+                    "is:open ((is:issue assignee:me) OR (is:pr author:me))");
+            panelControl.selectLastPanel();
+        });
+        return assigneePanel;
+    }
+
+    public MenuItem createRecentlyUpdatedPanel(){
+        MenuItem recentlyUpdatedPanel = new MenuItem("Recently Updated issues");
+        recentlyUpdatedPanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Create Recently Updated issues panel");
+            generatePanelWithNameAndFilter("Recently Updated issues", "assignee:me updated:<48");
+            panelControl.selectLastPanel();
+        });
+        return recentlyUpdatedPanel;
+    }
+
+    public MenuItem createMilestonePanel(){
+        MenuItem recentlyUpdatedPanel = new MenuItem("Current Milestone");
+        recentlyUpdatedPanel.setOnAction(e -> {
+            logger.info("Menu: Panels > Create Current Milestone panel");
+            generatePanelWithNameAndFilter("Current Milestone", "milestone:curr sort:status");
+            panelControl.selectLastPanel();
+        });
+        return recentlyUpdatedPanel;
+    }
+
+    public FilterPanel generatePanelWithNameAndFilter(String panelName, String filterName){
+        FilterPanel panelAdded = panelControl.addPanelAt(panelControl.getPanelCount());
+        panelAdded.setPanelName(panelName);
+        panelAdded.setFilterByString(filterName);
+        return panelAdded;
+    }
+}
```

Comment:
Yes, I think it's possible. But I think the method
`generatePanelWithNameAndFilter` would be better to be inside
`PanelControl`? This is to prevent the same issue with board currently
where board methods is unnecessarily inside menu instead of in its own
class, which in panel case would be `PanelControl`.

In reusing the code, should I do it in a separate issue once this PR is
done?
On Dec 14, 2015 8:04 AM, "Damith C. Rajapakse" <notifications@github.com>
wrote:

> In src/main/java/ui/PanelMenuCreator.java
> <https://github.com/HubTurbo/HubTurbo/pull/1141#discussion_r47454700>:
>
> > +        MenuItem recentlyUpdatedPanel = new MenuItem("Current Milestone");
> > +        recentlyUpdatedPanel.setOnAction(e -> {
> > +            logger.info("Menu: Panels > Create Current Milestone panel");
> > +            generatePanelWithNameAndFilter("Current Milestone", "milestone:curr sort:status");
> > +            panelControl.selectLastPanel();
> > +        });
> > +        return recentlyUpdatedPanel;
> > +    }
> > +
> > +    public FilterPanel generatePanelWithNameAndFilter(String panelName, String filterName){
> > +        FilterPanel panelAdded = panelControl.addPanelAt(panelControl.getPanelCount());
> > +        panelAdded.setPanelName(panelName);
> > +        panelAdded.setFilterByString(filterName);
> > +        return panelAdded;
> > +    }
> > +}
>
> @HansNewbie <https://github.com/HansNewbie> can code be reused across
> this PR and code for auto-creating boards, because creating boards involve
> creating panels? If not reuse, at least some consistency between the two?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1141/files#r47454700>.
>